libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/MapboxTripStarterTest.kt

```diff
@@ -0,0 +1,162 @@
+package com.mapbox.navigation.core.trip
+
+import com.mapbox.android.core.permissions.PermissionsManager
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.trip.session.TripSessionState
+import com.mapbox.navigation.testing.MainCoroutineRule
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineScope
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertThrows
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class, ExperimentalCoroutinesApi::class)
+class MapboxTripStarterTest {
+    @get:Rule
+    var coroutineRule = MainCoroutineRule()
+
+    private val testCoroutineScope: TestCoroutineScope = coroutineRule.coroutineScope
+    private val sut = MapboxTripStarter().apply {
+        // TODO this solution is temporary https://mapbox.atlassian.net/browse/NAVAND-1061
+        coroutineScopeProvider = { testCoroutineScope }
```

Comment:
Is this really necessary? I read the comments and still don't get it. `MainCoroutineRule` already mocks `Dispatchers.Main` for you. Without this line the tests still pass.

Reply:
After I moved `checkState` into `setLocationPermissionGranted`, it appears not needed anymore. But I'd like to leave this because this [issue is still valid](https://mapbox.atlassian.net/browse/NAVAND-1061)

If your object throws an exception inside a coroutine scope it will be swallowed
```
        coroutineScope.launch {
            error("the test will log this but will still succeed")
        }
```

Reply:
I moved the `checkState` call to demonstrate

Reply:
I still don't see any differences. There is a failing test now. If I remove this line, the test still fails.

Reply:
Hmm yeah I'm not fixing anything here. The TestCoroutineScope is swallowing the exception and I can't even find it in the uncaughtExceptions.

I can see it in the logs

Reply:
I didn't figure out how to catch the coroutine error for a test. It's being swallowed. So I've made `onStarterOptionsChanged` visible for testing so that the error has a unit test.

This had nothing to do with the actual behavior, maybe there is a better solution but I didn't find it.